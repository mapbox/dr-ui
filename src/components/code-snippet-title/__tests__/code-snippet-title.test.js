import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './code-snippet-title-test-cases.js';

describe('code-snippet-title', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.withLink.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.withLink;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.withToggle.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.withToggle;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
