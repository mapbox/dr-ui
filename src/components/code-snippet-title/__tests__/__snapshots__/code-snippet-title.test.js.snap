// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code-snippet-title Basic renders as expected 1`] = `
<div
  className="flex-parent-mm flex-parent--space-between-main-mm mb6"
>
  <div
    className="flex-child"
  >
    <div
      className="inline-block txt-bold mb6 color-gray-dark"
    >
      MainActivity.java
    </div>
  </div>
</div>
`;

exports[`code-snippet-title With a link renders as expected 1`] = `
<div
  className="flex-parent-mm flex-parent--space-between-main-mm mb6"
>
  <div
    className="flex-child"
  >
    <div
      className="inline-block txt-bold mb6 color-gray-dark"
    >
      MainActivity.java
    </div>
    <div
      className="inline-block ml12-mm"
    >
      <a
        className="unprose link"
        href="https://github.com/mapbox/"
        title="View MainActivity.java on GitHub"
      >
        <svg
          className="events-none icon inline-block align-t"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": 18,
              "width": 18,
            }
          }
        >
          <use
            xlinkHref="#icon-github"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          />
        </svg>
         View on GitHub
      </a>
    </div>
  </div>
</div>
`;

exports[`code-snippet-title With a toggle renders as expected 1`] = `
<div
  className="flex-parent-mm flex-parent--space-between-main-mm mb6"
>
  <div
    className="flex-child"
  >
    <div
      className="inline-block txt-bold mb6 color-gray-dark"
    >
      ViewController
    </div>
    <div
      className="inline-block ml12-mm"
    >
      <a
        className="unprose link"
        href="https://github.com/mapbox/"
        title="View ViewController on GitHub"
      >
        <svg
          className="events-none icon inline-block align-t"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": 18,
              "width": 18,
            }
          }
        >
          <use
            xlinkHref="#icon-github"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          />
        </svg>
         View on GitHub
      </a>
    </div>
  </div>
  <div
    className="flex-child"
  >
    <div
      className="relative "
    >
      <fieldset>
        <div
          aria-required={true}
          className="toggle-group round-full bg-blue py3 px3"
          data-test="some-id-input"
          id="some-id"
          role="radiogroup"
        >
          <label
            className="toggle-container "
          >
            <input
              autoFocus={false}
              checked={true}
              name="some-id"
              onChange={[Function]}
              type="radio"
              value="swift"
            />
            <div
              className="txt-s py3 toggle--white toggle--active-blue toggle"
            >
              Swift
            </div>
          </label>
          <label
            className="toggle-container "
          >
            <input
              autoFocus={false}
              checked={false}
              name="some-id"
              onChange={[Function]}
              type="radio"
              value="objectiveC"
            />
            <div
              className="txt-s py3 toggle--white toggle--active-blue toggle"
            >
              Objective-C
            </div>
          </label>
        </div>
      </fieldset>
      <div
        role="alert"
      />
    </div>
  </div>
</div>
`;
